pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,564 @@
+		// Request object is alive, so ensure it has the DNS finalizer.
+		if !controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) {
+			controllerutil.AddFinalizer(instance, reconcileSSHDFinalizerDNS)
+			if err = r.client.Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// Request object is being deleted.
+		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) && r.route53 != nil {
+			r.SetSSHDStatus(instance, "Deleting DNS aliases", cloudingressv1alpha1.SSHDStateFinalizing)
+			if err = r.deleteDNSRecords(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Remove the DNS finalizer and update the request object.
+			controllerutil.RemoveFinalizer(instance, reconcileSSHDFinalizerDNS)
+			if err = r.client.Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Halt the reconciliation.
+		return reconcile.Result{}, nil
+	}
+
+	// List ConfigMaps with SSH keys
+	configMapList := &corev1.ConfigMapList{}
+	selector, err := metav1.LabelSelectorAsSelector(&instance.Spec.ConfigMapSelector)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if err = r.client.List(context.TODO(), configMapList,
+		client.InNamespace(instance.Namespace),
+		&client.MatchingLabelsSelector{Selector: selector}); err != nil {
+		r.SetSSHDStatusError(instance, reqLogger, "Failed to list config maps with SSH keys", err)
+		return reconcile.Result{}, err
+	}
+	// Sort ConfigMaps by name for the purpose of creating
+	// stable Volume and VolumeMount lists in the Deployment.
+	sort.Slice(configMapList.Items, func(i, j int) bool {
+		return configMapList.Items[i].Name < configMapList.Items[j].Name
+	})
+
+	// Install Deployment
+	foundDeployment := &appsv1.Deployment{}
+	deployment := newSSHDDeployment(instance, configMapList)
+	deploymentName, err := client.ObjectKeyFromObject(deployment)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if err = r.client.Get(context.TODO(), deploymentName, foundDeployment); err != nil {
+		if errors.IsNotFound(err) {
+			// Create a new Deployment.
+			r.SetSSHDStatusPending(instance, reqLogger, "Creating deployment")
+			if err := controllerutil.SetControllerReference(instance, deployment, r.scheme); err != nil {
+				r.SetSSHDStatusError(instance, reqLogger, "Failed to set deployment controller reference", err)
+				return reconcile.Result{}, err
+			}
+			if err = r.client.Create(context.TODO(), deployment); err != nil {
+				if errors.IsAlreadyExists(err) {
+					return reconcile.Result{Requeue: true}, nil
+				}
+				r.SetSSHDStatusError(instance, reqLogger, "Failed to create deployment", err)
+				return reconcile.Result{}, err
+			}
+		} else {
+			return reconcile.Result{}, err
+		}
+	} else {
+		// Deployment exists, check if it's updated.
+		if !reflect.DeepEqual(foundDeployment.Spec, deployment.Spec) {
+			// Specs aren't equal, update and fix.
```

Comment:
What does "equal" mean here? Why should they be? In other words, I can see that the line above is comparing the equality (or inequality, to be more precise) but why is that being done?

Reply:
This is typically done when the operator itself has a configuration change, and needs to update the already installed deployment with a new option.

Reply:
I can see that, @cblecker, but what is expected to be changed at this point, and why?

Reply:
We don't know yet? We know this this operator owns this object, why wouldn't we enforce the spec of it?

Reply:
As I've come to understand it, this idiom both applies _intentional_ changes to the spec but also reverts _unintentional_ changes from, for example, attempted manual edits to the resource.